### PR TITLE
Apply ioredis, and add has redis cluster condition

### PR DESCRIPTION
**Pull request by galaxybing**

### CODE REVIEW

**Comment by vampolo** [comment]
Came across this problem and the fact that `redis` package does not support clustering as mentioned in https://github.com/NodeRedis/node_redis/issues/574 . Any hope for this to be merged? 
If not I'll just put this PR in a fork and move along.

**Comment by niftylettuce** [comment]
I'll take care of this now

**Comment by niftylettuce** [comment]
@galaxybing @vampolo could you guys please add docs to README.md to show people how to do cluster support once I release this version?

**Comment by vampolo** [comment]
Thank you very much. The version in this PR is a bit outdated but should do the job. I'll open a PR with a bump in version for ioredis

**Comment by niftylettuce** [comment]
@vampolo I will bump deps now

**Comment by niftylettuce** [comment]
@DaAwesomeP @dead-horse could you please grant me NPM owner on GitHub?

`npm owner add niftylettuce koa-redis`

File: index.js [hunk 1]
```diff
@@ -16,6 +16,7 @@
 var EventEmitter = require('events').EventEmitter;
 var debug = require('debug')('koa-redis');
 var redis = require('redis');
+var ioredis = require("ioredis");
```
Comment (dead-horse) [inline index.js]: I'd like to switch to ioredis and remove redis, then we can release a major version.

**Comment by dead-horse** [comment]
@niftylettuce added.

File: index.js (hunk 1 above)
Comment (niftylettuce) [reply index.js]: I'll take care of this now

**Comment by niftylettuce** [comment]
I've added the docs now and will be pushing soon for major release

**Comment by niftylettuce** [comment]
I've just released v4.0.0 which has support for Clusters and Sentinels.  Please see updated docs and examples at <https://github.com/koajs/koa-redis#koa-redis>

File: index.js (hunk 1 above)
Comment (niftylettuce) [reply index.js]: Done

**Pull request merged**